addons/stock_account/models/product.py

```diff
@@ -695,13 +695,29 @@ def _svl_replenish_stock(self, description, products_orig_quantity_svl):
             ('lot_id', '!=', False),
             ], ['product_id', 'location_id', 'lot_id'], ['quantity:sum'],
             having=[('quantity:sum', '>', 0)])
+        no_lot_quants = self.env['stock.quant']._read_group([
```

Comment:
I would filter out product without `lot_valuated` to save one potential `_read_group`

Reply:
From my tests, if the tracking is "none", `lot_valuated` is False, so using `self.product_variant_ids.filtered("lot_valuated").ids` seems to be enough

Reply:
You will do the read_group anyway even if you do not have any lot_valuated product but it's ok we don't call this method a lot

Reply:
Updated it